continuationsPlugin/src/main/scala/continuations/ContinuationsPlugin.scala

```diff
@@ -74,3 +96,141 @@ class ContinuationsPhase extends PluginPhase:
     tree match
       case Apply(_, _) => returnsContextFunctionWithSuspendType(tree)
       case _ => false
+
+  def hasOnlySuspendParam(tree: DefDef)(using Context): Boolean =
+    tree.termParamss match
+      case List(Nil, param :: Nil) => isSuspendType(param.tpe)
+      case _ => false
+
+  def getAllSubTrees(tree: Tree)(using Context): List[Tree] = {
+    val deepFolder: DeepFolder[ListBuffer[tpd.Tree]] =
+      new DeepFolder[ListBuffer[Tree]]((subTrees, tree) => subTrees += tree)
+
+    /*
+     * Needed for `case Inlined` because DeepFolder.foldOver for Inlined doesn't use `call` tree so the Inlined
+     * is not being unwrapped (???)
+     */
+    @tailrec
+    def recurse(trees: List[Tree], buf: ListBuffer[Tree]): List[Tree] =
+      trees match
+        case Nil =>
+          buf.toList
+        case Inlined(call, _, _) :: rest =>
+          recurse(call :: rest, buf)
+        case t :: rest =>
+          val (inlined, notInlined) =
+            deepFolder.apply(ListBuffer[Tree](), t).partition {
+              case _: Inlined => true
+              case _ => false
+            }
+          recurse(inlined.toList ++ rest, notInlined ++ buf)
+
+    recurse(tree.toList, ListBuffer[Tree]())
```

Comment:
💡 If we move the transformation to a class like suggested above, we can move this method there.

Reply:
got rid of this method, it was helping us to deal with `Inlined` and also look for nested `suspendContinuation` calls. but I guess atm we don't need to worry about nested ones, just look for top level, and that simplified the code for now https://github.com/47deg/TBD/pull/52/commits/1a5f9ef8011ed082a480c0da6afa4bae1cc71944#diff-fade65ebf959155389570d62d274801c2f7607f965a273661e5f776636d6393fR142